### PR TITLE
Reserve `sarif` crate

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

As discussed offline, this reserves the `sarif` name on crates.io.

## 🔍 What does this change?

- Adds boilerplate to define a crate in HASH

## 🛡 What tests cover this?

- /s The crate currently has 💯% line coverage _and_ branch coverage

## ❓ How to test this?

- Verify the links in the README and potential typos as this will be published.
- /s Run the full test suite, this may take up to 0 seconds

### CODE REVIEW

File: libs/sarif/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+version = "0.0.0-reserved"
+authors = ["HASH"]
+edition = "2021"
+license = "MIT OR Apache-2.0"
+description = "Representation of the SARIF specification in Rust"
+documentation = "https://docs.rs/sarif"
+readme = "README.md"
+repository = "https://github.com/hashintel/hash/tree/main/libs/sarif"
+keywords = ["sarif", "codeql", "serde", "static-analysis", "tools"]
+categories = ["data-structures", "development-tools", "parser-implementations", "parsing"]
```
Comment (indietyp) [inline libs/sarif/Cargo.toml:12]: is `parser-implementations` fitting here? AFAIK we do not plan to parse anything, just output (and transform) into valid SARIF.

File: libs/sarif/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+`sarif` is a crate to represent the [SARIF specification](https://sarifweb.azurewebsites.net) in Rust.
+
+## ⚠️ Disclaimer ⚠️
+
+This crate does not ship with any functionality and is only a name reservation to stop potential name squatting. In the future the actual crate will be published under this name, for the current (incomplete) implementation of the crate please visit the linked repository.
+
+## Contributors
+
+`sarif` was created by [Tim Diekmann](https://github.com/TimDiekmann) and [Bilal Mahmoud](https://github.com/indietyp) to be used to improve DX around creating Pull Requests. It is being developed in conjunction with [HASH](https://hash.dev/) as an open-source project. We gratefully accept external contributions and have published a [contributing guide](https://github.com/hashintel/hash/blob/main/CONTRIBUTING.md) that outlines the process. If you have questions, please reach out to us on our [Discord server][discord]. You can also report bugs [directly on the GitHub repo](https://github.com/hashintel/hash/issues/new?assignees=TimDiekmann%2Cindietyp&labels=A-sarif%2CC-bug&template=bug-report-sarif.yml).
```
Comment (indietyp) [inline libs/sarif/README.md:27]: ❤️

**Review by indietyp — CHANGES_REQUESTED** [review]
A couple of questions; otherwise looks good. This also needs:

* CODEOWNERS (@nonparibus)
* renovate needs to be configured.

File: libs/sarif/Cargo.toml (hunk 1 above)
Comment (TimDiekmann) [reply libs/sarif/Cargo.toml:12]: Probably leaving that out is good
```suggestion
categories = ["data-structures", "development-tools", "parsing"]
```

**Comment by TimDiekmann** [comment]
Thanks, @indietyp, I added the labeller and renovate configuration. As a drive-by I also added `deer` to the publish patterns (we forgot to add it) and grouped the Rust toolchain bumps in renovate to avoid 5 PRs per week.

**Review by indietyp — APPROVED** [review]

**Pull request merged**